src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -41,6 +41,13 @@ def load_arguments(self, _):
     identity_role_type = CLIArgumentType(options_list=['--role'], help="Role name or id that will be assigned to the managed identity")
     extended_json_format_type = CLIArgumentType(options_list=['--handle-extended-json-format', '-j'], action='store_true', is_preview=True,
                                                 help='Support to handle extended template content including multiline and comments in deployment')
+    deployment_name_type = CLIArgumentType(options_list=['--name', '-n'], required=True, help='The deployment name.')
+    deployment_create_name_type = CLIArgumentType(options_list=['--name', '-n'], required=False, help='The deployment name. Default to template file base name')
+    management_group_id_type = CLIArgumentType(options_list=['--management-group-id', '-m'], required=True, help='The management group id.')
```

Comment:
Which kind of id is used here? When there is id as parameter suffix, we will expect GUID as parameter value.

Reply:
This is not necessarily a GUID, but it has been the official term for management group: https://docs.microsoft.com/en-us/azure/governance/management-groups/create.

Reply:
It is a kind of management group name, right? In addition, in bot cli and powershell commands, we don't have parameter named xx-id,  but we all use --name to identify the specific management group. 
For management group parameter here, can we just use --management-group as parameter name and in help message, we can use something like "The name of management group. For root management group, it should be the Azure Active Directory ID"?

Reply:
I'd rather keep the "management-group-id". If you look at the link I put above, the description said "The Management Group ID is the directory unique identifier that is used to submit commands on this management group." I think there's a discussion on bringing everything consistent to "management group ID".

Reply:
We should do name *or* ID, but not management-group because then it is ambiguous until I go to help.

I would prefer ID because of how we present this field to user's in the portal, but @Juliehzl if you feel strongly about using name because that's how we refer to it in PS/CLI, I'm ok with that.

cc @rthorn17 as FYI

Reply:
Trying to close on this one. @Juliehzl let us know what you think.

Reply:
Also, please note the equivalent PowerShell cmdlets `*-AzManagementGroupDeployment` have the parameter as 'ManagementGroupId'.

Reply:
@Juliehzl - can you provide guidance on this one?

Reply:
This is what we follows in https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#argument-naming-conventions.
And the following part is what I get for a management group using existing cli command. May I know which one is the value I should pass to your parameter if you are using `--management-group-id`? 
```
  {
    "displayName": "zunli",
    "id": "/providers/Microsoft.Management/managementGroups/zuhtestmg",
    "name": "zuhtestmg",
    "tenantId": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a",
    "type": "/providers/Microsoft.Management/managementGroups"
  }
```
From my understanding, one of the "name" and "id" part should work to specify a valid management group. If you can support both, you also should use --management-group with help message "Id or name of management group.“ If you only support name, it will be confused for me to use your commands.

Reply:
unfortunately, the property you need there is "name". The confusion is that in the Portal, that same property is called "ID". The question is what do we optimize for. We will take @zhoxing-ms approval as an acknowledgement that we can move forward with ID